### PR TITLE
[libjpeg-turbo] Upgrade to 2.1.2

### PR DESCRIPTION
**Pull request by ahojnnes**
Addresses issue: https://github.com/microsoft/vcpkg/issues/18731

### CODE REVIEW

File: ports/libjpeg-turbo/workaround_cmake_system_processor.patch [hunk 1]
```diff
@@ -1,13 +0,0 @@
-diff --git a/CMakeLists.txt b/CMakeLists.txt
```
Comment (JonLiu1993) [inline ports/libjpeg-turbo/workaround_cmake_system_processor.patch:1]: @ahojnnes, thanks for your PR, I noticed that you deleted this patch, but I can't find the upstream application of this patch, can you find out the commit that applied this patch？

File: ports/libjpeg-turbo/portfile.cmake [hunk 2]
```diff
@@ -38,7 +35,7 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
         jpeg8 WITH_JPEG8
 )
 
-vcpkg_configure_cmake(
+vcpkg_cmake_configure(
     SOURCE_PATH ${SOURCE_PATH}
     PREFER_NINJA
```
Comment (JonLiu1993) [inline ports/libjpeg-turbo/portfile.cmake:40]:
```suggestion
    SOURCE_PATH "${SOURC E_PATH}"
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
CI failed on x64-uwp triplet and arm-uwp triplet:
````
CMake Error at CMakeLists.txt:57 (string):
  string no output variable specified

````

I think maybe we should revert this patch:
_ports/libjpeg-turbo/workaround_cmake_system_processor.patch_

**Comment by ahojnnes** [comment]
@JonLiu1993 Addressed your comments. PTAL.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
All features are tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

**Comment by BillyONeal** [comment]
Thanks for the update and merging the tricky patches!

**Pull request merged**